daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/BitsCharsetDefinition.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.processors.charset
+
+/**
+ * These are the classes which must be dynamically loaded in order to add a charset implementation
+ * to Daffodil. All charsets must implement this class and be added to the 
+ * org.apache.daffodil.processors.charset.BitsCharsetDefinition file in 
+ * daffodil-io/src/main/resources/META-INF/services. Names are not case sensitive, so BitsCharsetDefinition
+ * that have the same name with different capitalization will return the most recent addition
+ */
+abstract class BitsCharsetDefinition {
+
+  def name(): String
```

Comment:
Actually, I believe what @stevedlawrence meant was that we should add a comment here saying that name() must return an all-upper-case string for the encoding to be findable by Daffodil.

Reply:
Yep, that's correct, sorry for the confusion. Eventually users are going to start implementing their own custom BitsCharsetDefinition and looking at this class for documentation on what should be returned. Making it clear that name() must return all uppercase is important.

Another option might be modify `SimpleNamedServiceLoader` to have some kind of option to specifiy this behavior, so that it just always capitalizes whatever name returns. Then users don't have to really care. Might need changes to support aliases anyways (not sure how best to handle that).